src/sink/elastic.rs

```diff
@@ -63,107 +67,339 @@ pub struct Elastic {
+                let bulk_res = req.send()?.into_response::<BulkResponse>()?;
+
+                if is_linked {
+                    // send out response events for each item
+                    // success events via OUT port
+                    // error   events via ERR port
+                    for (item, value) in bulk_res.iter().zip(event.value_iter()) {
+                        let item_res = match item {
+                            Ok(ok_item) => (
+                                OUT,
+                                build_bulk_success_data(
+                                    ok_item,
+                                    &id,
+                                    value.clone_static(), // uaarrghhh
+                                    event.origin_uri.as_ref(),
+                                ),
+                            ),
+                            Err(err_item) => (
+                                ERR,
+                                build_bulk_error_data(
+                                    err_item,
+                                    &id,
+                                    value.clone_static(), // uaarrrghhh
+                                    event.origin_uri.as_ref(),
+                                ),
+                            ),
+                        };
+                        if let (port, Ok(item_body)) = item_res {
+                            ress.push((item_body, port));
+                        } else {
+                            error!(
+                                "Error extracting bulk item response for document id {}",
+                                item.map_or_else(ErrorItem::id, OkItem::id)
+                            );
+                        }
+                    }
                 }
-                let d = start.elapsed();
-                let d = d.as_millis();
-                Ok(d as u64)
+                Ok(start.elapsed().as_millis() as u64)
             })();
 
-            let mut m = Value::object_with_capacity(2);
-            let cb;
-            if let Ok(t) = r {
-                if m.insert("time", t).is_err() {
-                    // ALLOW: this is OK
-                    unreachable!()
-                };
-                cb = CBAction::Ack;
-            } else {
-                // TODO update error metric here?
-                error!("Elastic search error: {:?}", r);
-                cb = CBAction::Fail;
+            let mut m = Value::object_with_capacity(1);
+            let cb = match &r {
+                Ok(time) => {
+                    if m.insert("time", *time).is_err() {
+                        // ALLOW: this is OK
+                        unreachable!()
+                    };
+                    CBAction::Ack
+                }
+                Err(e) => {
+                    // request failed
+                    // TODO update error metric here?
+                    println!("ES request failed: {:?}", e);
```

Comment:
lets use error so users can filter it

```suggestion
                    error!("ES request failed: {:?}", e);
```